Gems/Atom/RPI/Code/Tests/Image/StreamingImageTests.cpp

```diff
@@ -751,10 +751,13 @@ namespace UnitTest
         }
 
         // Validate retrieving a region of pixels
+        size_t outValuesIndex = 0;
         AZStd::vector<float> pixelValues(size.m_width * size.m_height);
         auto topLeft = AZStd::make_pair<uint32_t, uint32_t>(0, 0);
         auto bottomRight = AZStd::make_pair<uint32_t, uint32_t>(size.m_width, size.m_height);
-        RPI::GetSubImagePixelValues(imageAsset, topLeft, bottomRight, pixelValues);
+        RPI::GetSubImagePixelValues(imageAsset, topLeft, bottomRight, [&pixelValues, &outValuesIndex]([[maybe_unused]] const AZ::u32& x, [[maybe_unused]] const AZ::u32& y, const float& value) {
+            pixelValues[outValuesIndex++] = value;
```

Comment:
nit: You could always make it so `pixelValues` just reserves `size.m_width * size.m_height`, then you can just push values onto it instead of tracking `outValuesIndex`

Reply:
Good call, will update!